QQE-378.md

```diff
@@ -0,0 +1,44 @@
+# QQE-378 - HTTP module / REST use-cases - Development topic compression Brotli4j
+## Links
+Jira : [QQE-378](https://issues.redhat.com/browse/QQE-378)
+
+PR:  TODO
+## Scope of the testing
+ - Integration of the Brotli4J compression library with Quarkus, particularly in the context of REST endpoints using RestEasy Reactive.
+ - The test coverage needs to be defined for both JVM and native mode.
+ - To use Brotli4J compression we will need a custom Server configuration
+   * Utilize Vert.x http server options  to enable compression support.
+   * Include the compressor using Brotli4J options compression.
+   * Ensure that the server is configured to respect the Accept-Encoding header to automatically compress the response body with one of the supported compressions and send it back to the client.
+### Test scenarios 
+
+- Compression Enabled, Response Compressed
+   *    Set up a Quarkus application with the Brotli4J compression library
+   *    Configure the HTTP server to enable compression using io.vertx.core.http.HttpServerOptions
+   *    Add the Brotli compressor using `.addCompressor(io.netty.handler.codec.compression.StandardCompressionOptions.brotli())`
+   *    Send a request to a REST endpoint and verify that the response is compressed using Brotli
+- Compression Disabled, Response Not Compressed
+    *   Set up a Quarkus application with the Brotli4J compression library
+    *   Configure the HTTP server to disable compression using io.vertx.core.http.HttpServerOptions
+    *   Send a request to a REST endpoint and verify that the response is not compressed
+- JSON Payloads, Compression Works up to 8192 bytes
+    *   Create a JSON payload of varying sizes up to 8192 bytes
+    *   Send a request to a REST endpoint with the JSON payload and verify that the response is compressed
+  using Brotli.
+    * Create a JSON payload of more of 8192 bytes and check the error.
```

Comment:
I don't understand why it shouldn't work, please add links and explanation into the TP. Here https://github.com/quarkusio/quarkus/issues/34719#issuecomment-1638142179 Clement suggested more than 8192 is supported when using Vert.x directly?

Reply:
AKA if this will only work when Quarkus bumps to next major Vert.x version in a year time, you should document it as well. Otherwise it's strange to just expect errors without an explanation.

Reply:
Clement wrote, "Our Brotli support only works up to 8192 (1024*8) bytes." 
up to = until= not exceeding

Reply:
ok, but Quarkus relies on Vert.x that doesn't limit that. Saying 8192 is our number and we shall test it is crazy. It must be documented, there must be opened issue and linked to this TP. Thanks

Reply:
It's not directed to you, but I just don't want to accept this.

Reply:
Point is, some PR comment does not determine contract.

Reply:
But you are right, I read Clement's comment wrong. But I don't like that comment as we need documentation for users.